### PR TITLE
Android Emulator Script

### PR DESCRIPTION
**Pull request by rozPierog**
Script for starting/creating new Android emulator.

You can start it by `npm run android:emulator` or `yarn android:emulator`

Options:
 - Run one of the existing emulators
 - Create new emulator using already installed system image
 - Download new system image and then create new Emulator

Every emulator created by this script is based on Nexus 6P (1440x2560px (5.70")) to simplify configuration process.

**Android Studio is still required!**

`export JAVA_OPTS="-XX:+IgnoreUnrecognizedVMOptions --add-modules java.se.ee"` is needed because of [a bug with Java 1.9+](https://stackoverflow.com/questions/46402772/failed-to-install-android-sdk-java-lang-noclassdeffounderror-javax-xml-bind-a)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -15,7 +15,8 @@
     "test:ios": "cd native/iosTest; xcodebuild -workspace WatermelonTester.xcworkspace -scheme 'WatermelonTester' -destination 'name=iPhone 8' test",
     "swiftlint": "cd native; swiftlint",
     "ktlint": "cd native/androidTest; ./gradlew ktlint",
-    "ktlint:format": "cd native/androidTest; ./gradlew ktlintFormat"
+    "ktlint:format": "cd native/androidTest; ./gradlew ktlintFormat",
+    "android:emulator": "export JAVA_OPTS=\"-XX:+IgnoreUnrecognizedVMOptions --add-modules java.se.ee\"; node ./scripts/emulator.js"
```
Comment (radex) [inline package.json]: This breaks it for me:

```
 ~/Documents/Nozbe/WatermelonDB   android-emu-script ⇕ yarn android:emulator
yarn run v1.9.4
$ export JAVA_OPTS="-XX:+IgnoreUnrecognizedVMOptions --add-modules java.se.ee"; node ./scripts/emulator.js
Error: no se ha encontrado o cargado la clase principal java.se.ee
child_process.js:640
    throw err;
    ^

Error: Command failed: $ANDROID_HOME/tools/bin/sdkmanager --list | grep "system-images/"
Error: no se ha encontrado o cargado la clase principal java.se.ee

    at checkExecSyncError (child_process.js:600:11)
    at execSync (child_process.js:637:13)
    at Object.<anonymous> (/Users/radex/Documents/Nozbe/WatermelonDB/scripts/emulator.js:13:14)
    at Module._compile (internal/modules/cjs/loader.js:689:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:700:10)
    at Module.load (internal/modules/cjs/loader.js:599:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:538:12)
    at Function.Module._load (internal/modules/cjs/loader.js:530:3)
    at Function.Module.runMain (internal/modules/cjs/loader.js:742:12)
    at startup (internal/bootstrap/node.js:266:19)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
 ✘  ~/Documents/Nozbe/WatermelonDB   android-emu-script ⇕ java -version
java version "1.8.0_131"
Java(TM) SE Runtime Environment (build 1.8.0_131-b11)
Java HotSpot(TM) 64-Bit Server VM (build 25.131-b11, mixed mode)
 ~/Documents/Nozbe/WatermelonDB   android-emu-script ⇕ javac -version
javac 1.8.0_131
 ~/Documents/Nozbe/WatermelonDB   android-emu-script ⇕

```

File: scripts/emulator.js [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+const sdks = execSync(
+  `$ANDROID_HOME/tools/bin/sdkmanager --list | grep "system-images/" `,
+).toString()
+
+const askForEmu = [
+  {
+    type: 'list',
+    name: 'name',
+    message: 'Pick Emulator from list or add a new one',
+    pageSize: add(emulators.length, 4),
```
Comment (radex) [inline scripts/emulator.js]: lol :P `emulator.length + 4` is fine too :P

Comment (rozPierog) [reply scripts/emulator.js]: I was inspired by https://github.com/Nozbe/WatermelonDB/blob/master/scripts/publish.js#L36 😆

File: package.json (hunk 1 above)
Comment (rozPierog) [reply package.json]: Huh, can you try running simply `node ./scripts/emulator.js` to see if this works?

Comment (radex) [reply package.json]: It does work! Both opening existing and creating a new emulator works

**Comment by rozPierog** [comment]
@radex Can you try running `npm run android:emulator ` now?

**Comment by radex** [comment]
It works now, but spits out warnings:

```
yarn run v1.9.4
$ ./scripts/emulatorWithJavaCheck
javac: invalid flag: --version
Usage: javac <options> <source files>
use -help for a list of possible options
javac: invalid flag: --version
Usage: javac <options> <source files>
use -help for a list of possible options
? Pick Emulator from list or add a new one (Use arrow keys)
```

I think it's `-version`, not `--version`

**Comment by rozPierog** [comment]
Yeah my bad, with Java 10 both version work the same ¯\\\_(ツ)_/¯

**Comment by radex** [comment]
LGTM 👍

**Pull request merged**